### PR TITLE
Feature/patro23 47 add entity framework

### PR DESCRIPTION
**Pull request by a-lipert**

### CODE REVIEW

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]
How to use' instruction in README.MD is missing

File: src/modules/example/application/Example/GettingExamples/GetExamples.cs [hunk 1]
```diff
@@ -13,13 +14,27 @@ public record GetExamples();
...
 	/// <summary>
 	/// GetExamples query handler.
 	/// </summary>
 	/// <param name="query">Query.</param>
 	/// <returns>Paged list of examples.</returns>
-	public Task<PagedList<ExampleInfo>> Handle(GetExamples query)
+	public async Task<PagedList<ExampleInfo>> Handle(GetExamples query)
 	{
-		throw new NotImplementedException();
+		var examples = await this.exampleRepository.GetAll();
```
Comment (mateuszmuszynski) [inline src/modules/example/application/Example/GettingExamples/GetExamples.cs:35]: We shouldn't mix read and write model. Repository should be used in command handlers only. Please inject DbContext and use EF directly here.

File: src/modules/example/domain/ExampleAggregate.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@ private ExampleAggregate(Guid id, string name)
 	{
 		var exampleCreated = new ExampleCreatedDomainEvent(id, name);
 		this.Apply(exampleCreated, this.Handle);
+		this.CreatedOn = DateTime.Now;
```
Comment (mateuszmuszynski) [inline src/modules/example/domain/ExampleAggregate.cs:18]: We should store dates in Coordinated Universal Time. `DateTime.Now` -> `DateTime.UtcNow`

File: src/modules/example/infrastructure/Data/AppDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+using Intive.Patronage2023.Modules.Example.Domain;
+using Intive.Patronage2023.Modules.Example.Infrastructure.Data.DataConfiguration;
+using Microsoft.EntityFrameworkCore;
+
+namespace Intive.Patronage2023.Modules.Example.Infrastructure.Data
+{
+	/// <summary>
+	/// Database context.
+	/// </summary>
+	public class AppDbContext : DbContext
```
Comment (mateuszmuszynski) [inline src/modules/example/infrastructure/Data/AppDbContext.cs:10]: I would suggest to use a more specific name here (`ExampleDbContext), as we will use separate DB contexts in other modules.

**Review by mateuszmuszynski — CHANGES_REQUESTED** [review]

File: src/modules/example/infrastructure/Migrations/AppDbContextModelSnapshot.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                    b.Property<DateTime>("CreatedOn")
+                        .HasColumnType("datetime2");
+
+                    b.Property<string>("Name")
+                        .IsRequired()
+                        .HasColumnType("nvarchar(max)");
+
+                    b.HasKey("Id");
+
+                    b.ToTable("ExampleAggregates");
```
Comment (mateuszmuszynski) [inline src/modules/example/infrastructure/Migrations/AppDbContextModelSnapshot.cs:40]: Use `Example` as table name

File: src/modules/example/infrastructure/Data/DataConfiguration/ExampleAggregateEntityConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	/// </summary>
+	internal class ExampleAggregateEntityConfiguration : IEntityTypeConfiguration<ExampleAggregate>
+	{
+		/// <summary>
+		/// Configure method.
+		/// </summary>
+		/// <param name="builder">builder.</param>
+		public void Configure(EntityTypeBuilder<ExampleAggregate> builder)
+		{
+			builder.HasKey(x => x.Id);
```
Comment (mateuszmuszynski) [inline src/modules/example/infrastructure/Data/DataConfiguration/ExampleAggregateEntityConfiguration.cs:24]: Maybe we didn't stated is clearly, but the idea is that we would have separate DB context per module.

**Review by mateuszmuszynski — CHANGES_REQUESTED** [review]

File: src/modules/example/infrastructure/Intive.Patronage2023.Modules.Example.Infrastructure.csproj [hunk 6]
```diff
@@ -3,10 +3,14 @@
 	<PropertyGroup>
         <TargetFramework>net7.0</TargetFramework>
         <ImplicitUsings>enable</ImplicitUsings>
-        <Nullable>enable</Nullable>
```
[the remark this answers is not in the document]
Comment (mateuszgiedzinski) [reply src/modules/example/infrastructure/Intive.Patronage2023.Modules.Example.Infrastructure.csproj:6]: Ok I checked and we can ignore Nullable setting. It's by default enable in .net >=6

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: Intive.Patronage2023.sln [hunk 7]
```diff
@@ -33,6 +33,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "architecture", "architectur
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Intive.Patronage2023.Architecture.Tests", "tests\architecture\Intive.Patronage2023.Architecture.Tests.csproj", "{74167673-3DBF-442E-8FBB-D12B7116589F}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{D6E10887-57DB-47DC-893B-E886DA96C51E}"
```
[the remark this answers is not in the document]
Comment (a-lipert) [reply Intive.Patronage2023.sln]: Done.

File: README.md [hunk 8]
```diff
@@ -1 +1,23 @@
...
+
+2. To create a new migration you can use .NET CLI and execute the following command: dotnet ef migrations add ###NAME### --project <PathToProjectWhereIsDbContext> --startup-project <PathToMainApiProject> --context <ContextName> --output-dir <PathToMigrationsDir> --configuration Debug --no-build.
+
+Migration script example: 
+
+dotnet ef migrations add ExampleMigration --project .\src\modules\example\infrastructure --startup-project .\src\api\app --context ExampleDbContext --output-dir .\src\modules\example\infrastructure --configuration Debug --no-build.
+
+3. In order to create the database, apply database migrations by executing: dotnet ef database update. Optionally, you can add the specific name of migration that you want the database to be updated to: dotnet ef database update <MigrationName>.
+
+
```
[the remark this answers is not in the document]
Comment (a-lipert) [reply README.md]: Done.

File: src/api/app/Program.cs [hunk 9]
```diff
@@ -1,8 +1,7 @@
 using Intive.Patronage2023.Modules.Example.Api;
 
 var builder = WebApplication.CreateBuilder(args);
-
-builder.Services.AddExampleModule();
+builder.Services.AddExampleModule(builder.Configuration);
```
[the remark this answers is not in the document]
Comment (a-lipert) [reply src/api/app/Program.cs:5]: Done.

File: src/modules/example/infrastructure/Data/DataConfiguration/ExampleAggregateEntityConfiguration.cs [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+			builder.ToTable("Example", "Examples");
+			builder.Property(x => x.Id).HasColumnName("Id");
+			builder.Property(x => x.Name).HasColumnName("Name");
+			builder.Property(x => x.Name).HasMaxLength(256);
+			builder.Property(x => x.CreatedOn).HasColumnName("CreatedOn");
```
Comment (mateuszgiedzinski) [inline src/modules/example/infrastructure/Data/DataConfiguration/ExampleAggregateEntityConfiguration.cs:23]: This HasColumnName and HasMaxLength are fluent methods.  What does mean? 
Fluent method means that they returns object that called it. So if you call builder.Property(x => x.Name).HasColumnName("Name") you probably can add also call for HasMaxLength(256) like
`builder.Property(x => x.Name).HasColumnName("Name").HasMaxLength(256);`

**Review by mateuszgiedzinski — APPROVED** [review]
In my opinion code review can be approved conditionally.

File: src/modules/example/infrastructure/Data/DataConfiguration/ExampleAggregateEntityConfiguration.cs (hunk 10 above)
Comment (a-lipert) [reply src/modules/example/infrastructure/Data/DataConfiguration/ExampleAggregateEntityConfiguration.cs:23]: Got it. You're right. Should I change that and commit again?

**Pull request merged**